07-routing-gossip.md

```diff
@@ -285,6 +285,7 @@ The following `address descriptor` types are defined:
          onion service addresses; Encodes:
          `[32:32_byte_ed25519_pubkey] || [2:checksum] || [1:version]`, where
          `checksum = sha3(".onion checksum" | pubkey || version)[:2]`.
+   * `6`: WebSocket port; data = `[2:port]` (length 2)
```

Comment:
I'd rather we allow specification of an IP as well here. Though maybe that interacts poorly with the hostname gossip thing? There may likely end up being nodes that would love to allow websocket connections, but sit behind cloudflare/nginx/whatever when doing so, whereas native LN protocol obviously doesn't proxy through a web proxy. Thus, they'd want to be able to specify a different host for websockets. I suppose for cloudflare-proxied nodes (dear god people stop using cloudflare) you'd need a domain name, not an IP, and have to send the HTTP Host header.

Maybe this should be hostname-only? Or maybe two different gossips, one for hostnames and one for ips?

Reply:
You basically iterate through all the other addresses, trying this port.  It's not ideal, but it seemed easier than some kind of specification which indicated *which* other descriptors it applied to...

Reply:
Yea, totally fair, I think my broader point here is that we really should consider `wss` a required part of this, because supporting websockets so browser clients can connect but then not support `wss` so browser clients can't connect in a secure context seems like it almost completely neuters the value here, sadly. For `wss` you need a hostname, ultimately, and I think we'd probably want to support a different hostname than the lightning node itself, as the `wss` part will probably be nginx or some other proxy, hence...I think we should just shove a hostname here :(

Reply:
+1 for wanting `wss`, we've created a websocket to tcp proxy server for connecting in a browser, but we can't do `ws` unfortunately.